service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/UpdateGceInstanceResourceZoneAttributesStep.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.workspace.service.resource.controlled.cloud.gcp.gceinstance;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CREATE_GCE_INSTANCE_ZONE;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.db.DbSerDes;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ResourceKeys;
+
+public class UpdateGceInstanceResourceZoneAttributesStep implements Step {
+
+  private final ControlledGceInstanceResource resource;
+  private final ResourceDao resourceDao;
+
+  public UpdateGceInstanceResourceZoneAttributesStep(
```

Comment:
this look literally the same as updateNotebookResourceLocationAttributesStep. Can we pass in the resource type in the constructor to know which attributes to use and merge the two classes?